### ISSUE TITLE
Kernel cache on disk looks not working in CUDA 12

### ISSUE THREAD

**Issue by hajimen** [cupy/cupy#7391]
### Description

# Environment
- Windows 10 x64
`nvcc -V` output:
'''
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2023 NVIDIA Corporation
Built on Fri_Jan__6_19:04:39_Pacific_Standard_Time_2023
Cuda compilation tools, release 12.0, V12.0.140
Build cuda_12.0.r12.0/compiler.32267302_0
'''
Python 3.10.10
cupy-cuda12x 11.50

# Phenomenon
Kernel cache on disk looks not working. CuPy makes new cache files every time after a process launches.

# Cause
CUDA Toolkit's `nvrtcResult nvrtcGetPTX ( nvrtcProgram prog, char* ptx )` now returns PTX blob contains one-time random value, and cupy/cuda/compiler.py:_compile_with_cache_cuda() uses PTX blob's hash for key of cache.
https://github.com/cupy/cupy/blob/d0c1ac2d0aefd4daa32fed01e17fad47e4f1653e/cupy/cuda/compiler.py#L542

# Fix proposal
Change the line to `key_src = '%s %s %s' % (env, source, extra_source)`.

### To Reproduce

```py
import cupy as cp
from cupyx.scipy.ndimage import uniform_filter
uniform_filter(cp.eye(4))
```

### Installation

Wheel (`pip install cupy-***`)

### Environment

```
Python Version               : 3.10.10
CuPy Version                 : 11.5.0
CuPy Platform                : NVIDIA CUDA
NumPy Version                : 1.23.1
SciPy Version                : 1.9.0
Cython Build Version         : 0.29.33
Cython Runtime Version       : None
CUDA Root                    : C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.0
nvcc PATH                    : C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.0\bin\nvcc.EXE
CUDA Build Version           : 12000
CUDA Driver Version          : 12000
CUDA Runtime Version         : 12000
cuBLAS Version               : (available)
cuFFT Version                : 11001
cuRAND Version               : 10301
cuSOLVER Version             : (11, 4, 3)
cuSPARSE Version             : (available)
NVRTC Version                : (12, 0)
Thrust Version               : 200001
CUB Build Version            : 200001
Jitify Build Version         : 4a37de0
cuDNN Build Version          : None
cuDNN Version                : None
NCCL Build Version           : None
NCCL Runtime Version         : None
cuTENSOR Version             : None
cuSPARSELt Build Version     : None
Device 0 Name                : NVIDIA T400
Device 0 Compute Capability  : 75
Device 0 PCI Bus ID          : 0000:07:00.0
```

**Comment by leofang** [comment]
I believe this has been fixed in https://github.com/cupy/cupy/pull/7345 and will appear in CuPy v11.6+ 🙂

**Comment by hajimen** [comment]
> I believe this has been fixed in #7345 and will appear in CuPy v11.6+ 🙂

Got it. Thank you!
